### PR TITLE
Adding level zero adapter

### PR DESCRIPTION
**Pull request by bmyates**
WIP, not ready for review
Signed-off-by: Brandon Yates <brandon.yates@intel.com>

### CODE REVIEW

File: source/adapters/ur.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "ur.hpp"
+#include <cassert>
+
+// Controls tracing UR calls from within the UR itself.
+bool PrintTrace = [] {
+    const char *Trace = std::getenv("SYCL_PI_TRACE");
```
Comment (jandres742) [inline source/adapters/ur.cpp:15]: this belongs only on SYCL. I will submit a PR there to change it.

**Comment by kbenzie** [comment]
I believe this will need to wait for #363 to be actioned before it can be merged.

File: source/adapters/level_zero/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: MIT
```
Comment (omarahmed1111) [inline source/adapters/level_zero/CMakeLists.txt:2]: Just wanted to give an update that the license of the project have been updated in this https://github.com/oneapi-src/unified-runtime/pull/557. So, we should be able to update all the files license to the llvm one now.

**Comment by pbalcer** [comment]
L0 was adapter was finally added in a different PR :) Closing.

**Pull request closed without merging**